pkg/core/network.go

```diff
@@ -469,12 +469,37 @@ func (n *NetworkCommand) PartitionChain(ipset string) ([]*pb.Chain, error) {
 		return nil, nil
 	}
 
+	chains := make([]*pb.Chain, 0, 2)
+	var toChains, fromChains []*pb.Chain
+	var err error
+
+	if n.Direction == "to" || n.Direction == "both" {
+		toChains, err = n.getPartitionChain(ipset, "to")
+		if err != nil {
+			return nil, errors.WithStack(err)
+		}
+	}
+
+	if n.Direction == "from" || n.Direction == "both" {
+		fromChains, err = n.getPartitionChain(ipset, "from")
+		if err != nil {
+			return nil, errors.WithStack(err)
+		}
+	}
+
+	chains = append(chains, toChains...)
```

Comment:
It seems an empty Chain may appended here.

Reply:
checked the `Direction` in function `validNetworkPartition`, it must be one of `from`, `to` and `both`

Reply:
OK, I agree with you.